odl/solvers/functional/functional.py

```diff
@@ -923,7 +923,13 @@ def __init__(self, func, quadratic_coeff=0, linear_term=None):
                             ''.format(func))
 
         self.__functional = func
-        self.__quadratic_coeff = func.domain.field.element(quadratic_coeff)
+        quadratic_coeff = func.domain.field.element(quadratic_coeff)
+        if np.iscomplexobj(quadratic_coeff):
```

Comment:
This line can actually be ommited since implicitly if `imag != 0` then the object is complex.

```python
quadratic_coeff = func.domain.field.element(quadratic_coeff)
if quadratic_coeff.imag != 0:
    raise ...
self.__quadratic_coeff = quadratic_coeff.real
```